src/auth/src/credentials/service_account.rs

```diff
@@ -177,7 +332,14 @@ where
         let mut value = HeaderValue::from_str(&format!("{} {}", token.token_type, token.token))
             .map_err(CredentialError::non_retryable)?;
         value.set_sensitive(true);
-        Ok(vec![(AUTHORIZATION, value)])
+        let mut headers = vec![(AUTHORIZATION, value)];
+        if let Some(project) = &self.quota_project_id {
```

Comment:
Can you add a todo here to refactor the quota project logic. Right now we have these few lines to add the quota project in multiple places